deep_learning4e.py

```diff
@@ -338,64 +339,136 @@ def get_batch(examples, batch_size=1):
         yield examples[i: i + batch_size]
 
 
-def NeuralNetLearner(dataset, hidden_layer_sizes, l_rate=0.01, epochs=1000, batch_size=1,
-                     optimizer=stochastic_gradient_descent, verbose=None):
+class NeuralNetLearner(Learner):
```

Comment:
I think this is not the best way to implement NNs. Even though you improved and streamlined the process a bit, right now it is not very useful. In your `fit` function you both initialize the model *and* train it. It would be best if these two processes were separate. That way, we could initialize the architecture, fit it, and then fit it again if we need to. So, all the initialization stuff should go to `__init__` and all the training stuff in `fit`.

Reply:
Ok Antonis, I thought about when you said and I reported the implementation of NeuralNet learner and PerceptronLearner as they were as they represent only examples of how a neural network could be composed :)